src/xdealer_channel.cpp

```diff
@@ -0,0 +1,51 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille, Sylvain Corlay, Martin Renou          *
+* Copyright (c) 2016, QuantStack                                           *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#include "xdealer_channel.hpp"
+
+namespace xeus
+{
+
+    xdealer_channel::xdealer_channel(zmq::context_t& context,
+                                 const xeus::xconfiguration& config)
+        : m_socket(context, zmq::socket_type::dealer)
+        , m_end_point("")
+    {
+        m_end_point = xeus::get_end_point(config.m_transport, config.m_ip, config.m_shell_port);
```

Comment:
I think I have this (`config.m_shell_port`) from the inital design where we had a shell socket and control socket under the dealer channel. Not sure what port should we use in this case.